libs/events/import/api/src/lib/import/events-import.service.ts

```diff
@@ -0,0 +1,365 @@
+  })
+
+  // save in cache
+  const ttl = 8 * 60 * 60 * 1000 // 8 hours
+  await cacheManager.set('parsedEvents', importedEvents, ttl)
+  return importedEvents
+}
+
+const today = moment().startOf('day')
+const upcomingOnly = (XMLEvent: XMLEventType) => {
+  const startDate =
+    XMLEvent &&
+    XMLEvent.ActivityDates &&
+    typeof XMLEvent.ActivityDates[0] !== 'string' &&
+    XMLEvent.ActivityDates[0]?.ActivityDate &&
+    XMLEvent.ActivityDates[0]?.ActivityDate[0]?.['$'] &&
+    XMLEvent.ActivityDates[0]?.ActivityDate[0]?.['$'].startDate
+
+  if (!startDate) {
+    return
+  }
+  if (moment(startDate).isBefore(today)) return
+  return XMLEvent
+}
+
+const getImageUrl = (event: XMLEventType) => {
+  return (
+    (event?.ActivityMultimedia.length &&
+      event?.ActivityMultimedia[0]?.Images.length &&
+      event?.ActivityMultimedia[0]?.Images[0]?.Image?.length &&
+      event?.ActivityMultimedia[0]?.Images[0]?.Image[0].$?.url) ||
+    ''
+  )
+}
+
+const parseXMLEventToWpEvent = (XMLEvent: XMLEventType, source: Providers) => {
+  const activityDate =
+    (XMLEvent &&
+      XMLEvent.ActivityDates &&
+      typeof XMLEvent.ActivityDates[0] !== 'string' &&
+      XMLEvent.ActivityDates[0]?.ActivityDate &&
+      XMLEvent.ActivityDates[0]?.ActivityDate[0]?.['$'] &&
+      XMLEvent.ActivityDates[0]?.ActivityDate[0]?.['$']) ||
+    null
+
+  const startDate = activityDate?.startDate
+  const startTime = activityDate?.startTime
+  const endDate = activityDate?.endDate
+  const endTime = activityDate?.endTime
+
+  const start = `${startDate} ${startTime}`
+  const end = endDate ? `${endDate} ${endTime}` : null
+
+  const castInfo = XMLEvent.CastInformation[0].replace(/(<([^>]+)>)/gi, '')
+  const longDescription = XMLEvent.LongDescription[0].replace(/(<([^>]+)>)/gi, '')
+  const shortDescription = XMLEvent.ShortDescription[0].replace(/(<([^>]+)>)/gi, '')
+  const fallbackDescription = getFallbackDesc(source)
+  const parsedDescription = htmlToSlate(
+    longDescription || shortDescription || castInfo || fallbackDescription
+  )
+
+  // we need to add type: 'paragraph' because that's how it was done in WP in the past
+  parsedDescription[0] = {...parsedDescription[0], type: 'paragraph'} as any
+
+  const parsedEvent = {
+    id: XMLEvent['$'].originId,
+    modifiedAt: new Date(XMLEvent['$'].lastUpdate || ''),
+    name: XMLEvent.Title[0].replace(/(<([^>]+)>)/gi, ''),
+    description: parsedDescription,
+    status: EventStatus.Scheduled,
+    imageUrl: getImageUrl(XMLEvent),
+    externalSourceId: XMLEvent['$'].originId,
+    externalSourceName: source,
+    location: XMLEvent.Location[0]?.LocationAdress[0].replace(/(<([^>]+)>)/gi, '') || '',
+    startsAt: start ? new Date(start.trim()) : null,
+    endsAt: end ? new Date(end.trim()) : null
+  }
+
+  return parsedEvent
+}
+
+const fetchAndTransformImage = async (url: string): Promise<ArrayBufferUpload> => {
+  try {
+    const response = await axios.get(url, {responseType: 'arraybuffer'})
```

Comment:
Can we use the HTTP service and not Axios directly?

Reply:
axios is easier and it's not like I added it to the project, it was already there

Reply:
Yeah but this is not the Nextjs way, now you will have multiple axios instances and interceptors won't work as expected. Axios is only used for the `notify-jira.js`

Reply:
if I understand correctly, I need to install `@nestjs/axios` package and inject it in agenda basel constructor like this `private readonly httpService: HttpService,`

https://docs.nestjs.com/techniques/http-module

Also, how to make sure to get the image as `arraybuffer`?